runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpPauseState.java

```diff
@@ -0,0 +1,154 @@
+package net.runelite.client.plugins.xptracker;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.EnumMap;
+import java.util.Map;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.Skill;
+
+@RequiredArgsConstructor
+public class XpPauseState
+{
+	// Required parameters
+	private final XpTrackerConfig config;
+	private final Client client;
+	// Internal state
+	private final Map<Skill, XpPauseStateSingle> skillPauses = new EnumMap<>(Skill.class);
+	private final Map<Skill, XpPauseInfo> skillXps = new EnumMap<>(Skill.class);
+	private boolean cachedIsLoggedIn = false;
+	private Instant lastLoginInstant = null;
+	private boolean logoutTriggered = false;
+
+	/**
+	 * Tick the pause status for all skills, taking into account skill no-xp timeouts
+	 * and logout pause settings
+	 */
+	public void tick(Instant now)
+	{
+		tickXp(now);
+		tickLogout(now);
+	}
+
+	public boolean pauseSkill(Skill skill)
+	{
+		return findPauseState(skill).onManualPause();
+	}
+
+	public boolean unpauseSkill(Skill skill)
+	{
+		findInfo(skill).setLastChange(Instant.now());
+		return !findPauseState(skill).onUnPause();
+	}
+
+	public boolean isPaused(Skill skill)
+	{
+		return findPauseState(skill).isPaused();
+	}
+
+	// Private internal functions
+	private void tickXp(Instant now)
+	{
+		final int pauseAfterSeconds = Math.max(60, config.pauseSkillAfter() * 60);
```

Comment:
Why is this limited at 60 seconds?

Reply:
Also does this mean it can't be turned off? You have to accept that your xp counter is going to pause its timer after an hour?

Reply:
It's a lower bound of 60 seconds, since the user can select `0` as their config value, which isn't valid.

Reply:
As stated in the PR description, it is a change to default behavior that skills will be automatically paused if they don't receive any XP for a whole hour.

Reply:
I think you should just make the 0 valid value and make it default to keep current master behaviour and behaviour of all other clients.